src/Microsoft.ML.Transforms/WrappedPcaTransform.cs

```diff
@@ -0,0 +1,186 @@
+            {
+                _weightColumn = weightColumn;
+                _rank = rank;
+                _oversampling = oversampling;
+                _center = center;
+                _seed = seed;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var pairs = new List<(string input, string output)>();
+                foreach (var outCol in toOutput)
+                    pairs.Add((inputNames[((OutPipelineColumn)outCol).Input], outputNames[outCol]));
+
+                string weightColName = null;
+                if (_weightColumn != null)
+                {
+                    weightColName = inputNames[_weightColumn];
+                }
+
+                return new PcaEstimator(env, pairs.ToArray(), weightColName, _rank, _oversampling, _center, _seed);
+            }
+        }
+
+        /// <include file='doc.xml' path='doc/members/member[@name="Whitening"]/*'/>
+        /// <param name="input">The column to apply PCA to.</param>
+        /// <param name="rank">The number of components in the PCA.</param>
+        /// <param name="oversampling">Oversampling parameter for randomized PCA training.</param>
+        /// <param name="center">If enabled, data is centered to be zero mean.</param>
+        /// <param name="seed">The seed for random number generation.</param>
+        public static Vector<float> ToPrincipalComponents(this Vector<float> input,
+            int rank = 20,
+            int oversampling = 20,
+            bool center = true,
+            int seed = 0) => new OutPipelineColumn(input, rank, oversampling, center, seed);
+
+        /// <include file='doc.xml' path='doc/members/member[@name="Whitening"]/*'/>
+        /// <param name="input">The column to apply PCA to.</param>
+        /// <param name="weightColumn">The name of the weight column.</param>
+        /// <param name="rank">The number of components in the PCA.</param>
+        /// <param name="oversampling">Oversampling parameter for randomized PCA training.</param>
+        /// <param name="center">If enabled, data is centered to be zero mean.</param>
+        /// <param name="seed">The seed for random number generation.</param>
+        public static Vector<float> ToPrincipalComponents(this Vector<float> input,
+            Vector<float> weightColumn,
+            int rank = 20,
+            int oversampling = 20,
+            bool center = true,
+            int seed = 0) => new OutPipelineColumnWithWeights(input, weightColumn, rank, oversampling, center, seed);
```

Comment:
>weightColumn, rank, oversampling, center, seed [](start = 69, length = 46)

I would prefer Config class/structure as we have in Categorical/Term/NAReplace than passing multiple params